modules/launcher/src/main/java/org/mule/runtime/module/launcher/log4j2/LoggerContextConfigurer.java

```diff
@@ -207,18 +210,24 @@ private void addDefaultArtifactContext(MuleLoggerContext context) {
 
     if (context.getConfigLocation() == null) {
       addDefaultAppender(context, logFile.getAbsolutePath());
-    } else if (!hasFileAppender(context) && isUrlInsideDirectory(context.getConfigFile(), getMuleConfDir())
-        || context.getConfiguration().getAppenders().isEmpty()) {
-      // If the artifact logging is configured using the global config file and there is no file appender for the artifact, then
-      // configure a default one. Same if there is no appender configured
-      addDefaultAppender(context, logFile.getAbsolutePath());
-      removeConsoleAppender(context);
+    } else if (isUrlInsideDirectory(context.getConfigFile(), getMuleConfDir())) {
+      removeConsoleAppender(context, containerConsoleAppenderMatcher);
+      if (!hasFileAppender(context)) {
+        addDefaultAppender(context, logFile.getAbsolutePath());
+      }
+    } else {
+      // Removes Log4j added default console appender
+      removeConsoleAppender(context, defaultConsoleAppenderMatcher);
+
+      if (context.getConfiguration().getAppenders().isEmpty()) {
+        addDefaultAppender(context, logFile.getAbsolutePath());
+      }
     }
   }
 
-  private void removeConsoleAppender(LoggerContext context) {
+  private void removeConsoleAppender(LoggerContext context, Function<String, Boolean> appenderNameMatcher) {
     for (Appender appender : getRootLogger(context).getAppenders().values()) {
-      if (appender instanceof ConsoleAppender) {
+      if (appender instanceof ConsoleAppender && appenderNameMatcher.apply(appender.getName())) {
```

Comment:
move the `appender instanceof ConsoleAppender &&` code to each predicate, remove `Console` from this method name.